jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/test/java/org/jbpm/workbench/ht/client/editors/taskdetails/TaskDetailsPresenterTest.java

```diff
@@ -15,31 +15,160 @@
+        verify(viewMock).setDueDate(any());
+        verify(viewMock).setDueDateTime(any());
+        verifySetTaskDetails(actualOwner,
+                             status,
+                             priority);
+        verifyReadOnlyMode(0,
+                           false);
+    }
+
+    @Test
+    public void testUpdateDetails() {
+        String serverTemplateId = "serverTemplateId";
+        String containerId = "containerId";
+        Long taskId = 1L;
+        boolean isForLog = false;
+
+        TaskSelectionEvent event = new TaskSelectionEvent(serverTemplateId,
+                                                          containerId,
+                                                          taskId,
+                                                          "task",
+                                                          true,
+                                                          isForLog,
+                                                          "description",
+                                                          new Date(),
+                                                          "Completed",
+                                                          "actualOwner",
+                                                          2,
+                                                          1L,
+                                                          "processId");
+        presenter.onTaskSelectionEvent(event);
+
+        String description = "description";
+        Date dueDate = new Date();
+        int priority = 3;
+        presenter.updateTask(description,
+                             dueDate,
+                             priority);
+
+        verify(taskService).updateTask(serverTemplateId,
+                                       containerId,
+                                       taskId,
+                                       priority,
+                                       description,
+                                       dueDate);
+        final ArgumentCaptor<TaskRefreshedEvent> argument = ArgumentCaptor.forClass(TaskRefreshedEvent.class);
+        verify(taskRefreshedEvent).fire(argument.capture());
+        assertEquals(taskId,
+                     argument.getValue().getTaskId());
+    }
+
+    private void verifySetTaskDetails(String actualOwner,
+                                      String status,
+                                      String priority) {
+        verify(viewMock).setUser(actualOwner);
+        verify(viewMock).setTaskStatus(status);
+        verify(viewMock).setTaskPriority(priority);
+    }
 
-        verify(view).setTaskDescriptionEnabled(false);
-        verify(view).setDueDateEnabled(false);
-        verify(view).setDueDateTimeEnabled(false);
-        verify(view).setTaskPriorityEnabled(false);
-        verify(view).setUpdateTaskVisible(false);
+    private void verifyReadOnlyMode(int i,
+                                    boolean readOnly) {
```

Comment:
It's quite confusing to have the method which verifies read-only mode and then have an argument which determines whether it's read-only or not. 
Furthermore, the argument value opposes the argument name, having _false_ actually means read-only.

I would suggest removing the boolean argument and hard-code _false_ in all method calls inside. The method is used only with boolean being set to _false_ in tests anyway, so no ram is done and it's much clearer.